File: Shopee project/shopee/src/main/java/fptu/shopee/Model/UserOrderPendingPayment.java
package fptu.shopee.Model;

import fptu.shopee.Model.ManyToManyRelationshipTable.ListItemInOrderPendingPayment;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

import java.util.Date;
import java.util.Set;

@Entity
@Table(name = "user_order_pending_payment")
public class UserOrderPendingPayment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_order_pending_payment")
    private Long idOrderPendingPayment;

    @Column(name = "note_to_seller", length = 500)
    private String noteToSeller;

    @NotNull
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "order_datetime", nullable = false)
    private Date orderDatetime;

    @PrePersist
    protected void onCreate() {
        orderDatetime = new Date();
    }

    @ManyToOne
    @JoinColumn(name = "id_payment_method")
    private PaymentMethod paymentMethod;

    @ManyToOne
    @JoinColumn(name = "id_shop_voucher")
    private ShopVoucher shopVoucher;

    @ManyToOne
    @JoinColumn(name = "id_web_voucher")
    private WebVoucher webVoucher;

    @ManyToOne
    @JoinColumn(name = "id_delivery_voucher")
    private DeliveryVoucher deliveryVoucher;

    @ManyToOne
    @JoinColumn(name = "id_user")
    private User user;

    @OneToMany(mappedBy = "userOrderPendingPayment", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Set<ListItemInOrderPendingPayment> listItemsInOrderPendingPayment;


    public Long getIdOrderPendingPayment() {
        return idOrderPendingPayment;
    }

    public void setIdOrderPendingPayment(Long idOrderPendingPayment) {
        this.idOrderPendingPayment = idOrderPendingPayment;
    }

    public String getNoteToSeller() {
        return noteToSeller;
    }

    public void setNoteToSeller(String noteToSeller) {
        this.noteToSeller = noteToSeller;
    }

    public Date getOrderDatetime() {
        return orderDatetime;
    }

    public void setOrderDatetime(Date orderDatetime) {
        this.orderDatetime = orderDatetime;
    }
}

File: .NET/Project learn/test_asp.net_Shopee_by_SQl Server/Shopee_by_SQl Server_ASP.Net/Shopee_by_SQl Server_ASP.Net/Models/UserOrderPendingPayment.cs
using System;
using System.Collections.Generic;

namespace Shopee_by_SQl_Server_ASP.Net.Models;

public partial class UserOrderPendingPayment
{
    public int IdOrderPendingPayment { get; set; }

    public int IdUser { get; set; }

    public int IdPaymentMetod { get; set; }

    public int IdShopVoucher { get; set; }

    public int IdWebVoucher { get; set; }

    public int IdDeliveryVoucher { get; set; }

    public string? NoteToSeller { get; set; }

    public DateTime? OrderDatetime { get; set; }

    public virtual DeliveryVoucher IdDeliveryVoucherNavigation { get; set; } = null!;

    public virtual PaymentMethod IdPaymentMetodNavigation { get; set; } = null!;

    public virtual ShopVoucher IdShopVoucherNavigation { get; set; } = null!;

    public virtual User IdUserNavigation { get; set; } = null!;

    public virtual WebVoucher IdWebVoucherNavigation { get; set; } = null!;

    public virtual ICollection<ListItemInOrderPendingPayment> ListItemInOrderPendingPayments { get; set; } = new List<ListItemInOrderPendingPayment>();
}
